File: Problems.CPP/Problem_5.cpp
#include <iostream>
#include <vector>

using namespace std;

class Problem_5 {

private:

	vector<int> ReversedNumber(int number)
	{
		vector<int> reversedNumber;
		int remainder = 0;

		while (number > 0)
		{
			remainder = number % 10; // extract the last digit from number
			number = number/10; // Remove the last digit after extracting

			reversedNumber.push_back(remainder);
		}

		return reversedNumber;
	}

public:

	int ReadPositiveNumber(string Message) {

		int number;

		do {
			cout << Message;
			cin >> number;

		} while (number <= 0);

		return number;
	}

	void PrintResult(int number)
	{
		
		cout << "\nNumber after reversed : " << endl;

		for (int N : ReversedNumber(number))
		{
			cout << N << endl;
		}
	}
};

File: Problems.CSharp/Problems/0005-Reversed order/Problem_5.cs
namespace Problems.CSharp.Problems._0005_Reversed_order
{
    public class Problem_5
    {
        public int ReadPositiveNumber(string Message)
        {
            int number;

            do
            {
                Console.Write(Message);
            } while (!int.TryParse(Console.ReadLine(), out number) || number <= 0);

            return number;
        }

        private List<int> ReversedNumbe(int number)
        {
            int remainder = 0;
            List<int> reversedNumbers = new List<int>();
            
            while(number > 0)
            {
                remainder = number % 10; // Take last digit from number
                number = number / 10; // Remove last digit from number

                reversedNumbers.Add(remainder);
            }

            return reversedNumbers;
        }

        public void PrintResult(int number)
        {
            Console.WriteLine("The number after reversed is : ");

            foreach(int n in ReversedNumbe(number))
            {
                Console.WriteLine(n);
            }
        }
    }
}
